easybuild/easyblocks/n/neuron.py

```diff
@@ -60,85 +62,133 @@ def extra_options():
+                try:
+                    os.chdir(pwd)
+                except OSError as err:
+                    raise EasyBuildError("Failed to change back to %s: %s", pwd, err)
 
     def sanity_check_step(self):
         """Custom sanity check for NEURON."""
         shlib_ext = get_shared_lib_ext()
         binpath = os.path.join(self.hostcpu, 'bin')
         libpath = os.path.join(self.hostcpu, 'lib', 'lib%s.' + shlib_ext)
-        binaries = ["bbswork.sh", "hel2mos1.sh", "ivoc", "memacs", "mkthreadsafe", "modlunit", "mos2nrn",
-                    "mos2nrn2.sh", "neurondemo", "nocmodl", "oc"]
-        binaries += ["nrn%s" % x for x in ["gui", "iv", "iv_makefile", "ivmodl", "mech_makefile", "oc", "oc_makefile",
-                                           "ocmodl"]]
-        libs = ["ivoc", "ivos", "memacs", "meschach", "neuron_gnu", "nrniv", "nrnmpi", "nrnoc", "nrnpython", "oc",
-                "ocxt", "scopmath", "sparse13", "sundials"]
-
         # hoc_ed is not included in the sources of 7.4. However, it is included in the binary distribution.
         # Nevertheless, the binary has a date old enough (June 2014, instead of November 2015 like all the
         # others) to be considered a mistake in the distribution
+        binaries = []
         if LooseVersion(self.version) < LooseVersion('7.4'):
             binaries.append("hoc_ed")
 
-        custom_paths = {
+        extra_dirs = []
+        if LooseVersion(self.version) < LooseVersion('7.8.1'):
+            binaries = ["bbswork.sh", "hel2mos1.sh", "ivoc", "memacs", "mkthreadsafe", "modlunit", "mos2nrn",
+                        "mos2nrn2.sh", "neurondemo", "nocmodl", "oc"]
+            binaries += ["nrn%s" % x for x in ["gui", "iv", "iv_makefile", "ivmodl", "mech_makefile", "oc",
+                                               "oc_makefile", "ocmodl"]]
+            libs = ["ivoc", "ivos", "memacs", "meschach", "neuron_gnu", "nrniv", "nrnmpi", "nrnoc", "nrnpython",
+                    "oc", "ocxt", "scopmath", "sparse13", "sundials"]
+        # list of included binaries changed with cmake. See
+        # https://github.com/neuronsimulator/nrn/issues/899
+        else:
+            binaries = ["mkthreadsafe", "modlunit", "neurondemo", "nocmodl", "nrngui", "nrniv", "nrnivmodl",
+                        "nrnmech_makefile", "nrnpyenv.sh", "set_nrnpyenv.sh", "sortspike"]
+            libs = ["nrniv", "rxdmath"]
+            if self.with_python:
+                extra_dirs += [os.path.join("lib", "python"),
+                               os.path.join("lib", "python%(pyshortver)s", "site-packages")]
+
+        # (we can not pass this via custom_paths, since then the %(pyshortver)s template value will not be resolved)
+        self.cfg['sanity_check_paths'] = {
             'files': [os.path.join(binpath, x) for x in binaries] + [libpath % x for x in libs],
-            'dirs': ['include/nrn', 'share/nrn'],
+            'dirs': ['include', 'share/nrn'] + extra_dirs,
```

Comment:
No more nrn subfolder?

Reply:
ah, not with the CMake install. I guess I need to make this different for older versions.